js/translation-utils.js

```diff
@@ -0,0 +1,152 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+'use strict';
+const path = require('path');
+const jsonUtils = require('./jsonUtils.js');
+const glob = require('glob');
+const zluxUtil = require('./util');
+const acceptLanguageParser = require('accept-language-parser');
+
+const utilLog = zluxUtil.loggers.utilLogger;
+
+/* loadTranslations reads pluginDefinition.i18n.{lang}-{Country}.json files
+/  for a given plugin location.
+/ 
+/  Returns translation maps:
+/  {
+/    '{lang}-{Country}': {
+/      'key': 'value',
+/      ...
+/    },
+/    ...
+/  }
+*/
+function loadTranslations(pluginLocation) {
+  const translationMaps = {};
+  const relativePath = 'web/assets/i18n';
+  const filePrefix = 'pluginDefinition.i18n.';
+  const fileExt = '.json';
+  const pattern = path.join(
+    pluginLocation,
+    relativePath,
+    `${filePrefix}*${fileExt}`
+  );
+  const files = glob.sync(pattern, {});
+  for (const file of files) {
+    const basename = path.basename(file);
+    const languageCountry = basename.substring(
+      filePrefix.length,
+      basename.length - fileExt.length
+    );
+    let content;
+    try {
+      content = jsonUtils.parseJSONWithComments(file);
+    } catch (e) {
+      utilLog.warn(`Failed to parse translation file ${file}. File skipped`);
+      continue;
+    }
+    translationMaps[languageCountry] = content;
+  }
+  return translationMaps;
+}
+
+// translate translates Plugin Definition choosing most suitable translation map
+function translate(pluginDef, translationMaps, acceptLanguage) {
+  const availableTranslations = getAvailableTranslations(translationMaps);
+  const langCountry = acceptLanguageParser.pick(
+    availableTranslations,
+    acceptLanguage
+  );
+  if (!langCountry) {
+    return pluginDef;
+  }
+  const pluginDefClone = zluxUtil.clone(pluginDef);
+  const webContent = pluginDefClone.webContent;
+  const translationMap = translationMaps[langCountry];
+  translateObject(webContent, translationMap);
+  return pluginDefClone;
+}
+
+// getAcceptLanguageFromCookies builds acceptLanguage string
+// based on user preferences stored in cookies
+function getAcceptLanguageFromCookies(cookies) {
+  const prefix = 'org.zowe.zlux.zlux-app-manager.preferences';
+  const languageKey = `${prefix}.language`;
+  const localeKey = `${prefix}.locale`;
+  const language = cookies[languageKey];
+  const locale = cookies[localeKey];
+  if (language && locale) {
```

Comment:
"language" *could* be es-MX and locale *could* be US (for example) for a spanish speaker raised in Mexico, but currently living or visiting in the US.

Should either just use language, or check if it already has both language and locale before adding the locale.